policyengine_us/variables/gov/states/nm/tax/income/exemptions/nm_armed_forces_retirement_pay_exemption.py

```diff
@@ -0,0 +1,21 @@
+from policyengine_us.model_api import *
+
+
+class nm_armed_forces_retirement_pay_exemption(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "New Mexico armed forces retirement pay exemption"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://casetext.com/statute/new-mexico-statutes-1978/chapter-7-taxation/article-2-income-tax-general-provisions/section-7-2-513-effective-until-112025-exemption-armed-forces-retirement-pay"
+    defined_for = StateCode.NM
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.nm.tax.income.exemptions.armed_forces_retirement_pay
+        armed_forces_retirement_pay = tax_unit.members(
+            "military_retirement_pay", period
+        )
+
```

Comment:
Since the dependents' income are included in the modified gross income, I would assume it is also available to dependents. 
![image](https://github.com/user-attachments/assets/a27e1157-955d-45b6-bcbd-6fcdfcd7ce9a)

Reply:
It says "the taxpayer" so I'm not sure. Can you screenshot the form itself?

Reply:
[2022 PIT-ADJ line 23](https://taxsim.nber.org/historical_state_tax_forms/NM/2022/PITbook2022.pdf#page=56 )

Reply:
![image](https://github.com/user-attachments/assets/39f421fb-a3b9-495d-8573-530a144055fa)

Reply:
[These](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/0269092f-c177-496e-851f-6b1410a1a080/2021pit-adj-ins.pdf) are the instructions
![image](https://github.com/user-attachments/assets/f48b15b3-cde7-41b8-8be7-b278b0058433)

I dont see any language regarding dependents

Reply:
@PavelMakarchuk that's the instruction for 2021 form, this exemption starts in 2022. 
![image](https://github.com/user-attachments/assets/4d17e1ac-93cf-4004-9a3a-f6386e64c1f9)
![image](https://github.com/user-attachments/assets/384b02b8-c21a-43e1-a37f-c8fa712f8818)

Reply:
It refers to filer and spouse meaning dependents are likely not included. You could run the instructions through an LLM for a second opinion.

Reply:
I see, yes then we should include only head or spouse, thank you @hua7450